src/codegen/llvm/intrin_rule_rocm.cc

```diff
@@ -0,0 +1,78 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file intrin_rule_rocm.cc
+ */
+#ifdef TVM_LLVM_VERSION
+#if TVM_ROCM_RUNTIME
+
+#include <tvm/ir.h>
+#include <tvm/api_registry.h>
+#include <tvm/codegen.h>
+#include <string>
+#include "./llvm_common.h"
+
+namespace tvm {
+namespace codegen {
+namespace llvm {
+
+using namespace ir;
+
+// num_signature means number of arguments used to query signature
+template<unsigned id, int num_signature>
+inline void DispatchLLVMPureIntrin(const TVMArgs& targs, TVMRetValue* rv) {
```

Comment:
create a new file intrin_rule_llvm.h and move the two template functions into there so it is shared with intrin_rule_llvm.cc